### PR TITLE
[WIP] Patch Envelopes: support external binary blobs

### PR DESCRIPTION
**Pull request by uncleDecart**
## TL;DR

This is Work In Progress Pull request on adding external binary blobs. It has not been tested yet with controller, will provide updates here. 

## Things I wanted to discuss 

In #3435 @milan-zededa mentioned that it's not a good idea to increase types more than it's needed, so I wanted to discuss if it makes sense to move logic (i.e. `PatchEnvelopes` struct from types to zedrouter, since zedrouter uses it in the end, my main concern if patch envelopes would be used in LOC, which will require importing this.

## Background and high level implementation

Patch envelopes are nothing more but a data exposed to Edge Applications
by EVE in runtime. This is done by metadata server which is part of zedrouter.
There are two types of patch envelopes: inline and external.

Inline patch envelopes are small objects (10Kbytes max as of now) which
are part of `EdgeDevConfig`.

External patch envelopes are volumes, downloaded by EVE and exposed to
Edge Applications via metadata server. Volumes are handled by other `volumemgr`
service which means that with external patch envelopes we need to get updates
not only from `zedagent` but also from `volumemgr` and those updates are not in
sync. So in order to handle that we need to

1) Separate information sent by `zedagent` to what we actually expose in `zedrouter`
2) Implement structure which will handle async updates from `volumemgr` and `zedagent`
3) Switch metadata server to using structure implemented in 2)

### CODE REVIEW

**Comment by uncleDecart** [comment]
rework in #3495
